### DISCUSSION TITLE
[Feature request] [Layout animations] An option to skip entering/exiting animations if the element's parent is mounted/unmounted

### DISCUSSION THREAD

**Discussion by SimpleCreations** [software-mansion/react-native-reanimated#2513] [General]
I believe it is a fairly common use case when one would want elements to enter/exit with an animation only if their parent stays unchanged. Examples of such use cases may include:

- A list with existing items. When the user first navigates to the screen with the list, the items shouldn't enter with an animation. They should only animate when the user updates the list (adds/deletes items).
- Multiple tabs, each of which may display an animated callout box. Currently the box's entering/exiting animation would run every time the user switches tabs, while it should only run when the callout box appears/disappears while the user is viewing the tab in which it appears/disappears.

Here is the `AnimatedList` example with several items added to the list by default and animation duration increased to 1 second:

https://user-images.githubusercontent.com/12449725/136595998-a2626128-e29b-4bce-8e39-a7c9c4bd72bb.mp4

Because of this, currently the application of layout animations is severely limited without applying ugly workarounds (like, in the first case, storing a timestamp when the list item was added and passing `undefined` to the `entering` prop if the time is less than the time of mounting the parent).

**Comment by Noitidart** [comment]
I had a similar issue, I wanted to skip entering (exiting is fine). I couldn't find a workaround. Can you please share your workarounds?

I tried to condtionally set `layout` after mount but it didn't work:


```
function Tag() {
  const [layout, setLayout] = useState();

  useEffect(() => {
   setLayout(Layout.duration(3000));
  }, []);

  return <View layout={layout} />
}
```

This however does not ever animate, its' weird.

**Reply by exzos28** [reply]
any updates?

**Comment by richterd** [comment]
My current workaround looks like this:

```
//This is to save if the mounting/layouting took place
const isAnimationsEnabled = useRef(false)

//This is to check if the layouting took place and return the according animation
const ifEnabled = (animation: any): any => {
    return isAnimationsEnabled.current ? animation : undefined
  }

//This is enabling the animations after layouting
useLayoutEffect(() => {
  isAnimationsEnabled.current = true
}, [])

//This is the JSX code
return (
  <Animated.View entering={ifEnabled(FadeInDown)} exiting={ifEnabled(FadeOutDown)}>
   <Text>Content</Text>
  </Animated.View>
)
```

**Reply by abhijith-sunder** [reply]
modified it for my specific use case and it works great, thanks!

**Comment by abhijith-sunder** [comment]
It would be nice to see a "disabled" prop on the component so animations can be disabled/enabled as required.

**Comment by Aure77** [comment]
This feature to disable/cancel layout animation when unmount/unfocus screen would be useful, even more when using navigation (navigation transition animation + exiting animation are ugly and sometimes next screen become unresponsive).

**Reply by vovacodes** [reply]
Has anyone managed to come up with a workaround for the `exiting` animation in particular? It really is a huge UI issue when stuff starts to jump around on the screen you're navigating away from :(

**Reply by MarcusDelvecchio** [reply]
By setting parent view to display: 'none' the exiting animation can be avoided. Hope this helps.

**Comment by saravanakumargn** [comment]
This way we can do this.
```ts
const isScreenMountedRef = useRef(false);

useEffect(() => {
  isScreenMountedRef.current = true;
}, []);

<Animated.View entering={isScreenMountedRef.current ? SlideInUp : null}
            exiting={SlideOutUp}>
// code
</Animated.View>
```

**Comment by tobobo** [comment]
This workaround also works with [`useIsMounted`](https://github.com/juliencrn/usehooks-ts/blob/master/packages/usehooks-ts/src/useIsMounted/useIsMounted.ts) from [usehooks-ts](https://usehooks-ts.com/) out of the box.

**Comment by MarcusDelvecchio** [comment]
Still looking for a workaround to prevent exiting animations on unmount. Has anyone come up with a solution for this?

**Comment by bartlomiejbloniarz** [comment]
I am working on a solution in #5045. It should be getting merged soon

**Comment by MarcusDelvecchio** [comment]
In the meantime I stumbled on a work around for hiding exit animations by setting the parent view to `display: 'none'`. In doing this the child's exiting animation will not be shown and it will simply disappear.

**Comment by bartlomiejbloniarz** [comment]
#5045 has been merged. It will be available in the next release (and is currently available via nightly). [docs](https://docs.swmansion.com/react-native-reanimated/docs/layout-animations/layout-animation-config/)

**Comment by vnahornyi** [comment]
@bartlomiejbloniarz Can I skip all layout animations across the entire app with a single call in App.tsx if I’m using React Navigation with react-native-screens?

**Comment by bartlomiejbloniarz** [comment]
@vnahornyi There is no API for that. You could do that manually by unsetting the layout animation related props.
